### ISSUE TITLE
Extra rests are being added to exported MIDIs (Terranigma and other N-SPC games?)

### ISSUE THREAD

**Issue by Michpem** [vgmtrans/vgmtrans#155]
**Describe the bug**
As the title says, there is an annoying issue when exporting MIDI files from Terranigma SPCs as well as some other N-SPC games (I haven't tried many) like Super Metroid, Kirby's Dreamland 3 and Kirby Super Star.

Basically, what happens is that the program will, for some reason, add a rest every so often at the start of a measure, which ends up pushing all the notes forward a bit. The way it does it seems to be different for every game, however. This is what I've found so far:

- **Terranigma**: The program will add a 2-tick long rest starting from measure 8. It adds another at measures 34, 38, 42, 47, 50, 52... I can't draw any pattern, unfortunately. Tested with Dark Side Field and Final Battle.
- **Super Metroid**: Adds a 2-tick long rest starting from measure 3 in a 4-4-8 pattern? (3, 7, 11, 19, 23, 27...). Tested with both Brinstar songs.
- **Both Kirby games**: The added 2-tick rests seem to start from measure 2 onward. It doesn't seem to follow a pattern, as it's very erratic (sometimes a rest is added every 6 measures, sometimes every 7, sometimes every other measure, etc). Tested with Grassland 1 (KDL3) and Gourmet Race (KSS).

It's important to note that the rests stack with each other, so when a new one is added it becomes a 4-tick rest and it increases every time. Also, by tick I mean the shortest note possible in a MIDI file (1/192 of a note, I think?).

**To Reproduce**
Export a MIDI file from Terranigma or the aforementioned N-SPC games, but the bug might affect other games as well, seeing that all of the ones I tested are recognized as "NinSnes" by the program.

**Expected behavior**
MIDIs without those added rests.

**Desktop:**
 - OS: Windows 10
 - VGMTrans version: Interim Build 2017-02-25 (I can't compile the program to test the newer code updates...)

**Comment by sykhro** [comment]
I doubt it will solve anything but please try [the most recent build](https://ci.appveyor.com/project/mikelow/vgmtrans/build/artifacts)

**Comment by Michpem** [comment]
Hmm, yeah, I tried that version and it behaves the same.

**Comment by PhoenixBound** [comment]
I think this is the same problem I and some friends ran into.

In the game we were looking at (EarthBound), there were rests of 1 tick inserted every so often. Specifically, they happened every time a section of 8 tracks ended and a new one began. Coincidentally, the MIDI editor I use was bad at displaying the 1 tick rests, so the problem wasn't visible until a second tick appeared.

**Comment by Moppu** [comment]
I'm having the same issue with NSPC MIDI dumps.  Seems like the timing of those MIDI dumps includes some delay effects (these show up as "SDx" in OpenMPT, where x is 1-5, for anywhere from a 1/6 to 5/6 row time delay) on the dumped notes.   These values drift upward over time, and each drift is accompanied by a small, subtly-audible delay.  

I was using an MML conversion tool that was rounding these to integer values and ending up with just a big delay once in a while when it rolled over, and I eventually traced it back to these little timing issues with the original MIDI dump.

At least from what OpenMPT shows me (it may be doing some conversions here to make MIDIs look more like "tracker" style music formats), vgmt is writing out to 128-length MIDI patterns, and just by the look of how the values drift over time, I suspect some internal pattern length that isn't a power of two (255?) is causing these little slips in the timing conversion.  

The song I was testing with was the dark world theme ("Dark Golden Land") from aLTTP.  If you open the dump in OpenMPT you should be able to see what I'm talking about, on any channel.

I tried a couple Megaman X songs and one from Illusion of Gaia and did not see the issue.  The delay effects appear but do not drift over time.

I don't currently know of an easy way to wipe these out and re-time the song without doing a manual readjustment of every note.  Currently at a loss for reliable MIDI dumps of any NSPC song.

**Comment by KungFuFurby** [comment]
I have an example case that I can provide in the form of my own Super Famicompo entry (which was made using the original source variant of Kankichi-kun... which I manually adapted over to asar), [SoftWave](https://github.com/vgmtrans/vgmtrans/files/9863639/SoftWave-000.spc.zip)... though it has been redumped so that VGMTrans can detect it correctly, because my original SPC file starts before anything else is set up, and VGMTrans expects some pointers to be set up. Yes, it has some other shenanigans going on that also trip up VGMTrans, but the rest problem comes first, and I set my tempo pretty low, which means that this song was particularly vulnerable to any delays in the notation.

I tested this on VGMTrans Preview 1.1, and I have determined that it inserts at least one tick at a phrase boundary, if not more, right at the beginning. The most likely cause is either a rounding error or something else going on, though more investigation will be required.
